### PR TITLE
Feat token storage

### PR DESCRIPTION
**Pull request by AdamWillden**
Replaces PR #85. Based upon PR #89

See commit message:

>   feat(baseConfig): replace both `tokenPrefix` options with `tokenStorage` (breaking)
> 
> BREAKING CHANGE: Token prefixes were using another 'unrelated' option to make up the full storage keys. This was unnecessary, confusing and could have resulted in the same storage location being shared between both the refresh and access tokens. README updated to reflect current design.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -335,6 +322,27 @@ tokenName: 'token',
...
+clientId: false,
+// The property name used when storing the refresh token locally
+refreshTokenStorage: 'aurelia_refresh_token',
+      
+// If `refresh_token` is an object:
+// ------------------------------------------------------------
+
+// This is the property from which to get the token `{ "refresh_token": { "refreshTokenName" : '...' } }`
+refreshTokenName: 'refresh_token',
+// This allows the refresh token to be a further object deeper `{ "refresh_token": { "refreshTokenRoot" : { "refreshTokenName" : '...' } } }`
```
Comment (doktordirk) [inline README.md:342]: those samples confuse me as 'refresh_token' means different things in the default value and the sample, or?

File: README.md [hunk 2]
```diff
@@ -335,6 +322,27 @@ tokenName: 'token',
...
+// =====================
+
+// Option to turn refresh tokens On/Off
+useRefreshToken: false,
+// The option to enable/disable the automatic refresh of Auth tokens using Refresh Tokens
+autoUpdateToken: true,
+// Oauth Client Id
+clientId: false,
+// The property name used when storing the refresh token locally
+refreshTokenStorage: 'aurelia_refresh_token',
```
Comment (doktordirk) [inline README.md:335]: while it makes sense to have that here, i'd prefer to have it with the storage options

Comment (doktordirk) [inline README.md:335]: +1 for changing from prefix to full name

File: README.md (hunk 1 above)
Comment (AdamWillden) [inline README.md:342]: No that is the problem the next PR #92 fixes. at the moment there is no `refreshTokenProp`. It is fixed to a value of 'refresh_token'.

Comment (AdamWillden) [inline README.md:342]: It confused me at first but I wanted to make the README match the code even though I was about to change said code.

File: README.md (hunk 2 above)
Comment (AdamWillden) [inline README.md:335]: @doktordirk do you mean to move the storage options into a new comment group of their own? We can do that in another commit if that's okay?

Comment (doktordirk) [inline README.md:335]: +1

**Comment by AdamWillden** [comment]
**Note** I have not run this **or any related PRs** against a live application yet. I will try to do this later unless someone else can test it. I've gotta go do some house tasks!

**Comment by doktordirk** [comment]
merged
